### DISCUSSION TITLE
Vercel is serving my serverless functions’ sources instead of executing them

### DISCUSSION THREAD

**Discussion by zoul** [vercel/community#603] [Help] [answered]
Since yesterday Vercel is serving my serverless functions’ sources instead of executing them – when I `curl somewhere/api/endpoint`, I get the function source instead of the JSON it’s supposed to output.

* The function used to work fine for months before.
* The function is written in TypeScript.
* I am not using Next.js, just the magic `api` folder.
* No functional changes were made in the commits triggering the bad deployment (just content updates).
* The sources are served with a `video/mp2t` content-type, I guess it’s because of the `.ts` extension.
* Previous deployments continue to work fine.
* Redeploying does not help.
* The function runs fine locally with `vercel dev`.

What could be wrong?

**Comment by swarnava** [comment, marked as answer]
Hey there,

We recently pushed a fix for this. Can you try again and let us know if you are still facing this?

**Reply by zoul** [reply]
Thank you very much, the site works now.

**Reply by szul** [reply]
@swarnava Is this issue occurring again? I'm currently having the exact same problem that @zoul described. No changes to the application at all. Just a content change. Past deployments work fine. Deployments today are blowing up.

**Reply by swarnava** [reply]
Hi, there is no widespread issue reported. Can you create a case with deployment details? https://vercel.com/support/request
